core-services/chat/src/main/java/org/silverpeas/core/chat/ChatUsersRegistration.java

```diff
@@ -0,0 +1,105 @@
+ * "http://www.silverpeas.org/docs/core/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.chat;
+
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.chat.servers.ChatServer;
+import org.silverpeas.core.chat.servers.DefaultChatServer;
+import org.silverpeas.core.socialnetwork.relationship.RelationShipService;
+import org.silverpeas.core.util.logging.SilverLogger;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Registers into the remote Chat server an existing Silverpeas user with all of its relationships.
+ * If the user is already registered then nothing is performed.
+ * @author mmoquillon
+ */
+@Singleton
+public class ChatUsersRegistration {
+
+  @Inject
+  @DefaultChatServer
+  private ChatServer chatServer;
+  @Inject
+  private RelationShipService relationShipService;
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  /**
+   * Is the chat service enabled in Silverpeas?
+   * @return true if the chat service is enabled and then the users can be registered into the
+   * chat service. False otherwise.
+   */
+  public boolean isChatServiceEnabled() {
+    return chatServer.isAvailable();
+  }
+
+  /**
+   * Is the specified user already registered into the remote chat server?
+   * @param user the user to check the existence.
+   * @return true if the user has an account in the remote chat server, false otherwise.
+   * @throws ChatServerException if an error occurs while communicating with the remote chat
+   * server.
+   */
+  public boolean isAlreadyRegistered(final User user) throws ChatServerException {
+    return chatServer.isUserExisting(user);
+  }
+
+  /**
+   * Registers the specified user into the remote chat server. If the server is already registered
+   * into the remote chat server, then nothing is performed. If the chat service isn't enabled then
+   * nothing is performed.
+   *
+   * With the user registration, his relationships are also browsed in order to create each of them
+   * into the remote chat server. If a user targeted by a relationship hasn't yet an account in
+   * the chat server, then he's registered before creating the relationship in the server.
+   *
+   * Only non administrators can be registered into the remote chat server.
+   *
+   * @param user the user to register if not yet done.
+   * @throws ChatServerException if the registration fails.
+   */
+  public void registerUser(final User user) throws ChatServerException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.silverpeas.core.chat.ChatServerException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)